### PR TITLE
configUpdater: more strict about number types (#369)

### PR DESCRIPTION
**Pull request by nritsche**
configUpdater is now more strict about number types and does not accept `float` updates that were of a number type `integer` or `unsigned` before. `integer` and `unsigned` values are allowed to be used to update each other.

### CODE REVIEW

File: lib/core/configUpdater.cpp [hunk 1]
```diff
@@ -166,28 +168,19 @@ void configUpdater::rest_callback(connectionInstance &con, nlohmann::json &json)
...
-
-            if (it.value().type() == nlohmann::json::value_t::number_float &&
-                _init_values[uri].at(it.key()).type() != it.value().type()) {
-                std::string msg = fmt::format(
-                            "configUpdater: Update to endpoint '{}' contained" \
-                            " value '{}' of type float (expected type {}).",
-                            uri.c_str(), it.key().c_str(),
-                            _init_values[uri].at(it.key()).type_name());
+            // ...and for correct types:
+            // If type_name differs, they are of different type and we care.
```
Comment (andrerenard) [inline lib/core/configUpdater.cpp]: It might be worth saying what we are trying to achieve here in more detail.  i.e. something like "Reject changes in general type (string vs. number vs. option), and reject changes between (un)signed integers and floats, but allow both signed and unsigned integers to be interchanged"   As written it's fairly hard to parse from the comment what this does.

Comment (nritsche) [reply lib/core/configUpdater.cpp]: Done
